### PR TITLE
Fix datafolder creation

### LINKED ISSUE #1: BeforeQuitHook does not work properly
Opening a new Igor7 experiment and closing it right away creates the Procedures/Codebrowser/ DataFolder.
Therefore the save dialog is triggered every time.

Maybe the error is due to the new igor 7.01 because we already tried to solve it but I'm not sure.

### PR DESCRIPTION
**Pull request by ukos-git**
fix #1 

Problem was that `debugPrint()` was reading the global variable `enableDebug` and while trying to read, it also created the Package DataFolder.

While taking a look at the problem, I also enhanced the other global Variable handler functions.

### CODE REVIEW

File: procedures/CodeBrowser_utils.ipf [hunk 1]
```diff
@@ -83,10 +83,19 @@ Function GetScreenDimensions(rect)
 End
 
 // Outputs a debug message prefixed with the calling function of debugPrint
-Function debugPrint(msg)
+//
+// @param loglevel   debug level
+// 					 0: No Error
```
Comment (t-b) [inline procedures/CodeBrowser_utils.ipf]: @ukos-git Wrong indentation.

File: procedures/CodeBrowser_utils.ipf [hunk 2]
```diff
@@ -166,91 +175,134 @@ End
...
-	else
-		myVar = numValue
-		return 1
 	endif
+
+	myVar = numValue
+	if(numtype(numValue) == 0 && getGlobalVar(globalVar) != numValue)
+		// @todo There should be a test case for this
+		Abort "global variable content missmatch."
+	endif
```
Comment (t-b) [inline procedures/CodeBrowser_utils.ipf]: @ukos-git In which situation would that be triggered?

File: procedures/CodeBrowser_utils.ipf [hunk 3]
```diff
@@ -166,91 +175,134 @@ End
...
-		myVar = strValue
-		return 1
 	endif
+
+	myVar = strValue
+	if(cmpstr(getGlobalStr(globalVar), strValue))
+		// @todo quick check! There should be a test case for this
+		Abort "global string content missmatch."
+		return 0
+	endif
```
Comment (t-b) [inline procedures/CodeBrowser_utils.ipf]: Same as above.

File: procedures/CodeBrowser_utils.ipf [hunk 4]
```diff
@@ -166,91 +175,134 @@ End
...
 End
 
 Function killGlobalVar(globalVar)
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
-	NVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 
-	KillVariables/Z dfr:$globalVar
+	DFREF dfr = $pkgFolder
+	if(DataFolderRefStatus(dfr) != 1)
```
Comment (t-b) [inline procedures/CodeBrowser_utils.ipf]: Please use https://www.wavemetrics.com/code-snippet/datafolderexists-data-folder-references instead which has better semantics and works for free datafolders as well.

File: procedures/CodeBrowser_utils.ipf (hunk 2 above)
Comment (ukos-git) [reply procedures/CodeBrowser_utils.ipf]: We always want to get back, what we have written.

The numtype was just to quickly skip the NaN comparison.

Comment (t-b) [reply procedures/CodeBrowser_utils.ipf]: @ukos-git What I meant was: Against which bug/situation are you protecting against? From my understanding of the code the condition in the if clause can never be true. Or is that basically an assertion?

File: procedures/CodeBrowser_utils.ipf (hunk 1 above)
Comment (t-b) [reply procedures/CodeBrowser_utils.ipf]: @ukos-git I never found a syntax checker for doxygen comments.

File: procedures/CodeBrowser_utils.ipf (hunk 2 above)
Comment (ukos-git) [reply procedures/CodeBrowser_utils.ipf]: Right, it can never be true. Simpler to drop it here.

File: procedures/CodeBrowser_utils.ipf (hunk 1 above)
Comment (ukos-git) [reply procedures/CodeBrowser_utils.ipf]: i added a quick Doxyfile to the commit but it is not complaining. It is not even finding the functions correctly. Did I miss something.

Comment (t-b) [reply procedures/CodeBrowser_utils.ipf]: @ukos-git Is the awk script present and in the correct location?

Comment (ukos-git) [reply procedures/CodeBrowser_utils.ipf]: https://pastebin.com/kf1Gmgbz

script is present and executed but only struct is recognized. No functions.

Comment (t-b) [reply procedures/CodeBrowser_utils.ipf]: @ukos-git Fixed and pushed. You gotta use the example from the doxygen-filter-ipf repo, this has EXTRACT_ALL set to yes.

**Comment by ukos-git** [comment]
I could not find exactly what is the cause of the ws error. My `git --no-pager diff --check master..` does not complain for the commits, I have pushed. It's probably gone or I don't see it.

**Comment by ukos-git** [comment]
Spaces were copy-pasted from https://www.wavemetrics.com/code-snippet/datafolderexists-data-folder-references

**Pull request merged**